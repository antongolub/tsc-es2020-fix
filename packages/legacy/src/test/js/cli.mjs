import {dirname, resolve} from 'node:path'
import {fileURLToPath} from 'node:url'
import process from 'node:process'
import tempy from 'tempy'
import fse from 'fs-extra'

const __dirname = dirname(fileURLToPath(import.meta.url))
const fakeProject = resolve(__dirname, '../../../../core/src/test/fixtures/ts-project')
const cliPath = resolve(__dirname, '../../main/js/cli.mjs')

let processArgv
let nodePath

beforeAll(() => {
  processArgv = process.argv
  [nodePath] = processArgv
})

afterEach(() => {
  process.argv = processArgv
})

describe('cli', () => {
  it('applies tsc-esm-fix to target dir', async () => {
    const cwd = tempy.directory()
    await fse.copy(fakeProject, cwd)

    process.argv = [nodePath, cliPath, '--cwd', cwd, '--tsconfig', 'tsconfig.json', '--ext', '.mjs', '--target', 'target/es6']
    await import(cliPath)

    const indexPath = resolve(cwd, 'target/es6/index.mjs')
    expect(await fse.readFile(indexPath, 'utf8')).toMatchSnapshot()
  })
})
