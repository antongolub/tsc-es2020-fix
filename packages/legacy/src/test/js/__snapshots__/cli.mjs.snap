// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cli applies tsc-esm-fix to target dir 1`] = `
"import { foo } from './foo.mjs';
import './bar.mjs';
import * as e1def from 'e1/a/b/c/index.js';
import * as e1root from 'e1';
const { e1 } = e1def;
const { e1: e1x } = e1root;
export { e1, e1x };
export { m1 } from 'm1';
export { m1 as m1x } from 'm1/index.js';
export { e2 } from 'e2';
export { e2 as es3 } from 'e2/index';
export { e2 as es4 } from 'e2/alias';
export { e2foo } from 'e2/foo';
export { e2bar } from 'e2/bar-bundle';
export * from './foo.mjs';
export * from './baz/index.mjs';
export * from './q/u/x/index.mjs';
export const foobaz = foo + 'baz';
export { foo as foo1 } from './foo.mjs';
export { qux } from './qux.js/index.mjs';
export const dirname = /file:\\\\/\\\\/(.+)\\\\/[^/]/.exec(import.meta.url)[1];
export const filename = /file:\\\\/\\\\/(.+)/.exec(import.meta.url)[1];
console.log(foobaz);
"
`;
